proto/stream_plan.proto

```diff
@@ -454,6 +454,8 @@ message StreamFragmentGraph {
     bool is_singleton = 4;
     // Number of table ids (stateful states) for this fragment.
     uint32 table_ids_cnt = 5;
+    // Mark the dependent table id of this fragment.
+    uint32 dependent_table_id = 6;
```

Comment:
There might be multiple upstream `Materialize` in a fragment like hash join. 🤔

Reply:
Well, this's only work for chain's upstream, so it should alway only have one. Maybe should change to a better naming. 🤔

Reply:
We may still use `Vec` here in case we get multiple chains accidentally after some further changes. We can assert it contains at most one. 🤣

Reply:
BTW, the name `dependent_table_id` is a bit of confusing to me. Who depends on who and why? Is it `upstream_table_id` or something? Maybe rename it and/or add some explanations. 🥸